### PR TITLE
Add test to verify final sapling root in block header is updated.

### PR DESCRIPTION
**Pull request by bitcartel**

### CODE REVIEW

File: qa/rpc-tests/finalsaplingroot.py [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+        assert_equal(len(self.nodes[0].getblock("205")["tx"]), 2)
+        assert_equal(self.nodes[1].z_getbalance(saplingAddr1), Decimal("12.34"))
+        assert(root is not self.nodes[0].getblock("205")["finalsaplingroot"])
+
+        # Verify there is a Sapling output description (its commitment was added to tree)
+        result = self.nodes[0].getrawtransaction(mytxid, 1)
+        assert_equal(len(result["vShieldedOutput"]), 2)  # there is Sapling shielded change
+
+if __name__ == '__main__':
```
Comment (arielgabizon) [inline qa/rpc-tests/finalsaplingroot.py]: Nice job.
Might be nice to add a test where the sender is a shielded spend but not recipient
and see the root doesn't change. Perhaps I'll add a commit for that.

File: qa/rpc-tests/finalsaplingroot.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        recipients.append({"address": saplingAddr0, "amount": Decimal('20')})
+        myopid = self.nodes[0].z_sendmany(taddr0, recipients, 1, 0)
+        mytxid = wait_and_assert_operationid_status(self.nodes[0], myopid)
+
+        self.sync_all()
+        self.nodes[0].generate(1)
+        self.sync_all()
+
+        # Verify the final Sapling root has changed
+        blk = self.nodes[0].getblock("201")
```
Comment (LarryRuane) [inline qa/rpc-tests/finalsaplingroot.py]: Maybe index using the return from `getblockcount()`? That way (1) if more steps are added later, the test automatically adjusts, and (2) the reader is sure the test is referencing the most recent block.

File: qa/rpc-tests/finalsaplingroot.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+        self.sync_all()
+
+    def run_test(self):
+        # Activate Overwinter and Sapling
+        self.nodes[0].generate(200)
+        self.sync_all()
+
+        # Verfify genesis block contains null field for what is now called the final sapling root field.
+        blk = self.nodes[0].getblock("0")
+        assert_equal(blk["finalsaplingroot"], NULL_FIELD)
```
[the remark this answers is not in the document]
Comment (bitcartel) [reply qa/rpc-tests/finalsaplingroot.py]: Yes, this was mentioned on another ticket too...

File: qa/rpc-tests/finalsaplingroot.py (hunk 2 above)
Comment (bitcartel) [reply qa/rpc-tests/finalsaplingroot.py]: I considered it, but this test is small and it makes it easy to visually check what block we are checking.

File: qa/rpc-tests/finalsaplingroot.py (hunk 1 above)
Comment (bitcartel) [reply qa/rpc-tests/finalsaplingroot.py]: ACK

**Comment by bitcartel** [comment]
@zkbot r+

**Pull request merged**